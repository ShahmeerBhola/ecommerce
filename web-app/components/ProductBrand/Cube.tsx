import React from 'react';
import Link from 'next/link';
import WhiteBox from '../WhiteBox';
import { constructUrl } from '../../utils';
import isEmpty from 'lodash/isEmpty';

type Props = {
  brand: WordPress.ProductBrand;
  linkCategorySlug: string;
};

const Cube: React.FunctionComponent<Props> = ({ brand: { name, slug, image }, linkCategorySlug }) => {
  const BrandImage = (): JSX.Element => {
    const { src, alt } = image as WordPress.Image;
    return <img src={src} alt={alt} className="max-h-6" />;
  };

  const BrandBackupText = (): JSX.Element => (
    <span className="block uppercase font-bold text-xl text-black py-10 px-5">{name}</span>
  );

  return (
    <div className="flex flex-grow p-2 box-border w-full md:w-1/2 lg:w-1/3 max-w-auto md:max-w-1/2 lg:max-w-1/3">
      <WhiteBox className="flex flex-grow px-5 py-8 justify-center text-center">
        <Link href={constructUrl({ page: 'category', extra: `${linkCategorySlug}/${slug}` })}>
          <a className="flex justify-center items-center">{isEmpty(image) ? <BrandBackupText /> : <BrandImage />}</a>
        </Link>
      </WhiteBox>
    </div>
  );
};

export default Cube;
