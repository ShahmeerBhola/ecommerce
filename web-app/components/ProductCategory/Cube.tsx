import React from 'react';
import Link from 'next/link';
import Text from '../Text';
import PhotoBackground from '../PhotoBackground';
import { constructUrl } from '../../utils';
import isEmpty from 'lodash/isEmpty';

type Props = {
  category: WordPress.ProductCategory;
};

const Cube: React.FunctionComponent<Props> = ({ category: { slug, name, image } }) => {
  const wrapperClassNames = 'flex justify-end items-end h-full w-full p-5 text-right';

  const Name = (): JSX.Element => (
    <Text color="text-white" className="uppercase" size="text-4xl" weight="font-bold">
      {name}
    </Text>
  );

  const Photo = (): JSX.Element => {
    const { src } = image as WordPress.Image;
    return (
      <PhotoBackground image={src} className={wrapperClassNames}>
        <Name />
      </PhotoBackground>
    );
  };

  const ColorBackground = (): JSX.Element => (
    <div className={`${wrapperClassNames} bg-gray-80`}>
      <Name />
    </div>
  );

  return (
    <div className="h-cube w-full p-2 box-border md:w-1/2 lg:w-1/3">
      <Link href={constructUrl({ page: 'category', extra: slug })}>
        <a>{isEmpty(image) ? <ColorBackground /> : <Photo />}</a>
      </Link>
    </div>
  );
};

export default Cube;
