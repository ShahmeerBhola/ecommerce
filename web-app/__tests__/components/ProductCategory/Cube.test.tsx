import React from 'react';
import renderer from 'react-test-renderer';
import Cube from '../../../components/ProductCategory/Cube';

describe('ProductCategory Cube component', () => {
  test('it renders properly', () => {
    const tree = renderer
      .create(
        <Cube
          size="250"
          category={{
            id: 5,
            name: 'Test Category',
            description: 'This is our test category',
            slug: 'test-category',
            image: {
              id: 45,
              name: 'my-image.png',
              src: 'my-image.png',
              alt: 'my image',
            },
            menu_order: 1,
            count: 5,
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
