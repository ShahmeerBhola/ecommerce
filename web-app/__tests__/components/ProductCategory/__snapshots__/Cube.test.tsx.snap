// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCategory Cube component it renders properly 1`] = `
<div
  className="h-cube p-2 box-border 250"
>
  <a
    href="/category/test-category"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="bg-no-repeat h-full relative bg-cover"
      style={
        Object {
          "backgroundImage": "url(my-image.png)",
        }
      }
    >
      <h3
        className="font-primary font-bold text-xl absolute bottom-5 left-5 uppercase text-white"
      >
        Test Category
      </h3>
    </div>
  </a>
</div>
`;
